uspv/init.go

```diff
@@ -210,8 +203,10 @@ func (s *SPVCon) Connect(remoteNode string) error {
 	}
 	handShakeFailed := false // need to be in this scope to access it here
 	connEstablished := false
+	var con net.Conn
 	for len(listOfNodes) != 0 {
-		s.con, err = s.DialNode(listOfNodes)
+		con, err = s.DialNode(listOfNodes)
+		s.con = con
```

Comment:
Why would we return `net.Conn` from `DialNode` when we assign the `con` returned to `s.con` in the next line? It would make sense if we used `con` somewhere else, but we don't, so I think its better we assign a value to `s.con` directly inside `DialNode`